src/Raven.Studio/typescript/widgets/virtualGrid/columns/textColumn.ts

```diff
@@ -125,19 +125,19 @@ class textColumn<T> implements virtualColumn {
         }
 
         if (_.isArray(cellValue)) {
-            const value = generalUtils.escapeHtml(JSON.stringify(cellValue, null, 2));
+            const innerHtml = cellValue.length ? "&hellip;" : "&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;";
 
             return {
-                rawText: "[ ... ]",
+                rawText: `<span class="array-item">[${innerHtml}]</span> <sup>${cellValue.length}</sup>`,
                 typeCssClass: "token array"
             }
         }
 
         if (_.isObject(cellValue)) {
-            const value = generalUtils.escapeHtml(JSON.stringify(cellValue, null, 2));
+            const innerHtml = Object.keys(cellValue).length ? "&hellip;" : "&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;";
```

Comment:
for performance reasons please compute OBject.keys(cellValue).length 1 time.

Reply:
done